src/tests/components/Login.test.tsx

```diff
@@ -14,51 +15,105 @@ const renderComponent: ReactElement = (
+        render(renderComponent)
+        const x509PassAuthText = screen.getByText('x509 Certificate')
+        expect(x509PassAuthText).toBeInTheDocument()
 
-    const userPassPasswordInput = screen.getByPlaceholderText('Enter Password')
-    expect(userPassPasswordInput).toBeInTheDocument()
-    expect(userPassPasswordInput.innerHTML.length).toEqual(0)
+        const userPassAuthText = screen.getByText('Username / Password')
+        expect(userPassAuthText).toBeInTheDocument()
+    })
+})
+
+describe('Auth workflow', () => {
+    test('Userpass', () => {
+        render(renderComponent)
+        const userPassButton = screen.getByText('Username / Password')
+
+        act(() => {
+            userPassButton.click()
+        })
 
-    const userPassSignInButton = screen.getByText('Sign In')
-    expect(userPassSignInButton).toBeInTheDocument()
-    expect(userPassSignInButton.getAttribute('type')).toEqual('button')
-    expect(userPassSignInButton.getAttribute('aria-disabled')).toEqual('true')
+        const userPassUsernameInput =
+            screen.getByPlaceholderText('Enter Username')
+        expect(userPassUsernameInput).toBeInTheDocument()
+        expect(userPassUsernameInput.innerHTML.length).toEqual(0)
 
-    fireEvent.change(userPassUsernameInput, { target: { value: 'ddmlab' } })
-    fireEvent.change(userPassPasswordInput, { target: { value: 'secret' } })
+        const userPassPasswordInput =
+            screen.getByPlaceholderText('Enter Password')
+        expect(userPassPasswordInput).toBeInTheDocument()
+        expect(userPassPasswordInput.innerHTML.length).toEqual(0)
 
-    const userPassSignInButtonEnabled = screen.getByText('Sign In')
-    expect(userPassSignInButtonEnabled).toBeInTheDocument()
-    expect(userPassSignInButtonEnabled.getAttribute('type')).toEqual('submit')
-    expect(userPassSignInButtonEnabled.getAttribute('aria-disabled')).toEqual(
-        'false',
-    )
+        const userPassSignInButton = screen.getByText('Sign In')
+        expect(userPassSignInButton).toBeInTheDocument()
+        expect(userPassSignInButton.getAttribute('type')).toEqual('button')
+        expect(userPassSignInButton.getAttribute('aria-disabled')).toEqual(
+            'true',
+        )
 
-    expect(() => act(() => userPassSignInButtonEnabled.click())).not.toThrow()
+        fireEvent.change(userPassUsernameInput, { target: { value: 'ddmlab' } })
+        fireEvent.change(userPassPasswordInput, { target: { value: 'secret' } })
+
+        const userPassSignInButtonEnabled = screen.getByText('Sign In')
+        expect(userPassSignInButtonEnabled).toBeInTheDocument()
+        expect(userPassSignInButtonEnabled.getAttribute('type')).toEqual(
+            'submit',
+        )
+        expect(
+            userPassSignInButtonEnabled.getAttribute('aria-disabled'),
+        ).toEqual('false')
+
+        expect(() =>
+            act(() => userPassSignInButtonEnabled.click()),
+        ).not.toThrow()
```

Comment:
@gmatthews20 do you think we should actually throw an exception here? 
My reasoning: These tests are not connected to a real rucio server and in case we try to make requests to a non-existing or non-responding server, we should throw an error. So in this case we should expect to throw an error
What do you think?

Reply:
I am not sure, does this test not just test whether the button click works and not the response it receives?

Reply:
This is just testing that clicking sign-in should not throw an exception. Let's remove this as it is not the correct behavior or rather an anti-behavior. 

It should be enough to test that the userPassSignInButtonEnabled gets enabled once the username and password values are supplied. In the actual login workflow, clicking this button would make REST calls to the rucio auth server and testing that is beyond the scope of this file presently.

The Click tests make more sense the way you have implemented them above with the `voTab`s, where you check that clicking the tab does not raise any errors and then check that the UI elements are in the expected state.